backend-rust/src/indexer.rs

```diff
@@ -2260,7 +2266,9 @@ impl PreparedBakerEvent {
         &self,
         tx: &mut sqlx::Transaction<'static, sqlx::Postgres>,
         transaction_index: i64,
+        protocol_version: ProtocolVersion,
     ) -> anyhow::Result<()> {
+        let is_current_protocol_version_older_than_6 = ProtocolVersion::P6 > protocol_version;
```

Comment:
We could avoid the `ensure_conditional_affected_one_row` by using a range here
```suggestion
        let expected_affected_range = if ProtocolVersion::P6 > protocol_version {
            1..=1
        } else {
            // Prior to protocol version 7, removing a baker was effective after a cooldown period which was still allowing transactions updating other information on the baker, since we still remove the baker immediately for these blocks there might not be any row affected.
            0..=1
        };
```

Reply:
This is strictly for bakers I will alter the name a bit.